components/relevancy/src/lib.rs

```diff
@@ -79,3 +133,26 @@ pub struct InterestMetrics {
 }
 
 uniffi::include_scaffolding!("relevancy");
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_ingest() {
+        let top_urls = vec![
+            "https://food.com/".to_string(),
+            "https://hello.com".to_string(),
+            "https://pasta.com".to_string(),
+            "https://dog.com".to_string(),
+        ];
+        let relevancy_store =
+            RelevancyStore::new("file:test_store_data?mode=memory&cache=shared".to_owned())
+                .unwrap();
+        let mut result: HashMap<Interest, u32> = HashMap::new();
+        result.insert(Interest::Food, 2);
+        result.insert(Interest::Inconclusive, 1);
+        result.insert(Interest::Animals, 1);
+        assert_eq!(relevancy_store.ingest(top_urls).unwrap(), result);
```

Comment:
My favorite way to construct a hashmap is by iterating over an array.  I think you could rewrite this something like:

```
 assert_eq!(relevancy_store.ingest(top_urls).unwrap(), HashMap::from([
    (Interest::Food, 2),
    (Interest::Inconclusive, 1),
    (Interest::Animals, 1),
]);
```

I don't think this code needs to change, I just wanted to share the idiom.

Reply:
oooh i didn't know this could be done this way!